src/tools/fuzzing.h

```diff
@@ -1128,24 +1131,29 @@ class TranslateToFuzzReader {
     if (type != i32 && type != i64) return ret;
     if (!features.hasAtomics() || oneIn(2)) return ret;
     // make it atomic
+    auto* load = ret->cast<Load>();
     wasm.memory.shared = true;
-    ret->isAtomic = true;
-    ret->signed_ = false;
-    ret->align = ret->bytes;
-    return ret;
+    load->isAtomic = true;
+    load->signed_ = false;
+    load->align = load->bytes;
+    return load;
   }
 
-  Store* makeNonAtomicStore(Type type) {
+  Expression* makeNonAtomicStore(Type type) {
     if (type == unreachable) {
       // make a normal store, then make it unreachable
       auto* ret = makeNonAtomicStore(getConcreteType());
-      switch (upTo(3)) {
-        case 0: ret->ptr = make(unreachable); break;
-        case 1: ret->value = make(unreachable); break;
-        case 2: ret->ptr = make(unreachable); ret->value = make(unreachable); break;
+      if (auto* store = ret->dynCast<Store>()) {
```

Comment:
Nit: Might be simpler to return early if this condition is false rather than put the rest of the function in an if block.